test/teos/e2e/test_basic_e2e.py

```diff
@@ -511,7 +514,10 @@ def test_appointment_shutdown_teos_trigger_while_offline(run_bitcoind):
     assert appointment_info.get("appointment") == appointment.to_dict()
 
     # Shutdown and trigger
-    teosd_process.terminate()
+    rpc_client = RPCClient(config.get("RPC_BIND"), config.get("RPC_PORT"))
+    rpc_client.stop()
+    teosd_process.join()
```

Comment:
Is dealing with the pids necessary now? It feels like we should not need this anymore, just check that after stopping and before starting it again if we try to make a request it fails.

Also, we should be able to teardown from conftest using stop too.

Reply:
how do you know when it stopped? In principle it could take time. Since the main teosd process waits for all the subprocesses to stop before exiting, this seems the safe way to me.

Reply:
(Will change the teardown code in the conftest to use the RPCClient)

Reply:
Changed the teardown code in conftest; still using `teosd_process.join()`, though, both here and in the conftest.